node.go

```diff
@@ -346,32 +365,79 @@ func (node *Node) writeBytes(w io.Writer) cmn.Error {
 	return nil
 }
 
-func GetLeftNode(node *Node, t *ImmutableTree) *Node { return node.getLeftNode(t) }
-func (node *Node) getLeftNode(t *ImmutableTree) *Node {
-	if node.leftNode != nil {
-		return node.leftNode
+func GetLeftNode(node *Node, t *ImmutableTree) *Node { return node.getLeftNode(t, false) }
+func (node *Node) getLeftNode(t *ImmutableTree, updateVersion bool) *Node {
+	if node.leftNode == nil {
+		node.mtx.Lock()
+		defer node.mtx.Unlock()
+		if node.leftNode == nil {
+			node.leftNode = t.ndb.GetNode(node.leftHash)
+			if updateVersion {
+				node.leftNode.loadVersion = t.version
+			} else {
+				node.leftNode.loadVersion = node.leftNode.version - 1
```

Comment:
why minus 1 here

Reply:
as discussed, we can add more comments about loadVersion

Reply:
added